src/FSharp.AWS.DynamoDB/Expression/ProjectionExpr.fs

```diff
@@ -2,158 +2,163 @@
 
 open System
 open System.Collections.Generic
-open System.Reflection
 
 open Microsoft.FSharp.Reflection
 open Microsoft.FSharp.Quotations
 open Microsoft.FSharp.Quotations.Patterns
-open Microsoft.FSharp.Quotations.DerivedPatterns
-open Microsoft.FSharp.Quotations.ExprShape
 
-open Amazon.DynamoDBv2
 open Amazon.DynamoDBv2.Model
 
 open FSharp.AWS.DynamoDB.ExprCommon
 
 ///////////////////////////////
 //
-// Extracts projection expressions from an F# quotation of the form 
+// Extracts projection expressions from an F# quotation of the form
 // <@ fun record -> record.A, record.B, record.B.[0].C @>
 //
 //  c.f. http://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Expressions.AccessingItemAttributes.html
 //
 ///////////////////////////////
 
 type AttributeId with
-    member id.View(ro : RestObject, value : byref<AttributeValue>) : bool =
+
+    member id.View(ro: RestObject, value: byref<AttributeValue>) : bool =
         let notFound rest =
             // only raise if the last component of the document path is missing
-            if List.isEmpty rest then false
+            if List.isEmpty rest then
+                false
             else
-                sprintf "Document path '%s' not found." id.Name
-                |> KeyNotFoundException
-                |> raise
+                sprintf "Document path '%s' not found." id.Name |> KeyNotFoundException |> raise
```

Comment:
I’m not sold on one-lining these, having the `raise` at the end of the line is less obvious than having it at the bottom of the chain. I can experiment with winding back the infix expression length a bit but the likelihood is it will catch some & not others.

Reply:
yes 2 pipes on a line is def the limit
and I prefer the `raise` to stick out
as mentioned I'd map that to `raise (KeyNotFoundException $"...")` 
but sreading it over 3 lines doesnt really help anyone IMO other than forcing them to think, which is great of course